### PR TITLE
Fix latex

### PR DESCRIPTION
**Pull request by ZekNikZ**

### CODE REVIEW

File: src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java [hunk 1]
```diff
@@ -52,18 +58,19 @@
...
+            latexTemplate = new StringTemplate(readTemplate(templateName, "latex.tem"));
+            experienceTemplate = new StringTemplate(readTemplate(templateName, "experience.tem"));
+            fieldTemplate = new StringTemplate(readTemplate(templateName, "field.tem"));
+            separatorTemplate = new StringTemplate(readTemplate(templateName, "separator.tem"));
             categoryTemplates = new HashMap<>();
             for (CategoryType c : CategoryType.values()) {
-                categoryTemplates.put(c, new StringTemplate(Files.readString(Path.of(templateDirectory, templateName, c.getTemplateFileName() + ".tem"))));
+                categoryTemplates.put(c, new StringTemplate(readTemplate(templateName, c.getTemplateFileName() + ".tem")));
             }
+
```
Comment (brooklynnStone1) [inline src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java:74]: Should this be logging instead of print?

**Review by brooklynnStone1 — COMMENTED** [review]
I would just change the System.out.println and System.out.err to LOG.INFO and LOG.ERR in ResumeTemplate at lines 73 and 84. Otherwise it looks good!

**Comment by ZekNikZ** [comment]
> I would just change the System.out.println and System.out.err to LOG.INFO and LOG.ERR in ResumeTemplate at lines 73 and 84. Otherwise it looks good!

I misunderstood some stuff about the classloader, which is why I didn't originally make these logged, but you are right, they should be logged.

**Review by nnazo — APPROVED** [review]

File: src/main/java/to/us/resume_builder/presentation/CreateOrOpenDialog.java [hunk 2]
```diff
@@ -95,9 +98,14 @@ public CreateOrOpenDialog() {
...
             try {
-                Desktop.getDesktop().open(new File(Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("userguide.pdf")).getFile()));
-            } catch (IOException e) {
+                if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
+                    Desktop.getDesktop().browse(new URI("http://resume-builder.us.to/userguide.html"));
+                } else {
+                    JOptionPane.showMessageDialog(this, "Could not open user guide.");
+                }
+            } catch (URISyntaxException | IOException e) {
                 e.printStackTrace();
```
Comment (MicahSchiewe) [inline src/main/java/to/us/resume_builder/presentation/CreateOrOpenDialog.java]: Shouldn't this be logged?

File: src/main/java/to/us/resume_builder/business/server_connect/request/BasicRequest.java [hunk 3]
```diff
@@ -18,56 +18,58 @@
...
-     * 
-     * @param arguments The arguments to send the request with, in a [Name1, Val1,
-     *                  Name2, Val2...] format.
-     * @return The http response from the request this class handles
-     * @throws InterruptedException If the operation is interrupted
-     * @throws IOException          If an I/O exception occurs in sending or
-     *                              receiving.
-     */
-    public final HttpResponse<T> sendRequest(String... arguments) throws IOException, InterruptedException {
+	/** The client leveraged by all subclasses. */
```
Comment (MicahSchiewe) [inline src/main/java/to/us/resume_builder/business/server_connect/request/BasicRequest.java:23]: Didn't realize that you were editing the BasicRequest class. I made some changes here in addition to fixing Javadoc compile errors (changing some stuff about this class that irked me), and it got onto master before this. I hope it doesn't make too painful a merge conflict to resolve.

File: src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java [hunk 4]
```diff
@@ -74,10 +83,28 @@
...
+        URL path = Thread.currentThread().getContextClassLoader().getResource("templates/" + templateName + "/" + fileName);
+        try (InputStream in = path.openStream();
+             BufferedReader reader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8))) {
+            String line = "";
+            while ((line = reader.readLine()) != null) {
+                template.append(line).append("\n");
+            }
+        }
+
+//        System.out.println(template.toString());
```
Comment (MicahSchiewe) [inline src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java]: Probably want to delete that comment.

**Review by MicahSchiewe — COMMENTED** [review]
Looks pretty good, apart from the couple of issues I pointed out.

File: src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java [hunk 5]
```diff
@@ -52,18 +59,20 @@
      * @param templateName The name of the template to load.
      */
     ResumeTemplate(String templateName) {
-        String templateDirectory = ApplicationConfiguration.getInstance().getString("templates.directory");
+        final Logger LOGGER = Logger.getLogger(ResumeTemplate.class.getName());
```
Comment (MicahSchiewe) [inline src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java:62]: Aren't loggers supposed to be static? Not a big deal, though.

**Review by MicahSchiewe — APPROVED** [review]
Looks good to me.

File: src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java (hunk 5 above)
Comment (ZekNikZ) [reply src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java:62]: You can't access static variables from the constructors of enums. Very weird thing, I know

Comment (MicahSchiewe) [reply src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java:62]: Huh. Learn something new every day.

File: src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java [hunk 6]
```diff
@@ -74,10 +83,26 @@
                 categoryTemplates.put(c, new StringTemplate(""));
             }
 
+            LOGGER.severe("Could not read template files for template " + templateName);
+
             e.printStackTrace(); // TODO: display error message
```
Comment (MicahSchiewe) [inline src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java]: There's a TODO here. That important?

File: src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java [hunk 7]
```diff
@@ -74,10 +83,26 @@
...
         }
     }
 
+    private String readTemplate(String templateName, String fileName) throws IOException {
+        StringBuilder template = new StringBuilder();
+        URL path = Thread.currentThread().getContextClassLoader().getResource("templates/" + templateName + "/" + fileName);
+        try (InputStream in = path.openStream();
```
Comment (MicahSchiewe) [inline src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java:95]: It lets you have a try block with no catch? Mine's never let me do that.

**Review by MicahSchiewe — COMMENTED** [review]
A couple more comments, but nothing big.

File: src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java (hunk 7 above)
Comment (ZekNikZ) [reply src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java:95]: A try-with-resources clause only requires a `catch` block if an exception can be thrown within it.

Comment (MicahSchiewe) [reply src/main/java/to/us/resume_builder/business/export_LaTeX/ResumeTemplate.java:95]: Well, fascinating. Carry on.

**Review by MicahSchiewe — COMMENTED** [review]
Looks good to merge.

**Pull request merged**